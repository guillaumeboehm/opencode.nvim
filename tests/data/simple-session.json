[
  {
    "properties": {
      "info": {
        "id": "msg_9cf8f64de0016tbfTQqWMydbdr",
        "role": "user",
        "sessionID": "ses_6345495f3ffeHrPDfQKfwjff2i",
        "time": { "created": 1760123905246 }
      }
    },
    "type": "message.updated"
  },
  {
    "properties": {
      "part": {
        "id": "prt_9cf8f64df001zPv6k1dQQefQQb",
        "messageID": "msg_9cf8f64de0016tbfTQqWMydbdr",
        "sessionID": "ses_6345495f3ffeHrPDfQKfwjff2i",
        "text": "only answer the following, nothing else:\n\n1",
        "type": "text"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "part": {
        "id": "prt_9cf8f64e0001Hje5NS5LoHtqfu",
        "messageID": "msg_9cf8f64de0016tbfTQqWMydbdr",
        "sessionID": "ses_6345495f3ffeHrPDfQKfwjff2i",
        "synthetic": true,
        "text": "Called the Read tool with the following input: {\"filePath\":\"/Users/cam/tmp/a/a-empty.txt\"}",
        "type": "text"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "part": {
        "id": "prt_9cf8f64e0002LnLCHbA2pQoFLJ",
        "messageID": "msg_9cf8f64de0016tbfTQqWMydbdr",
        "sessionID": "ses_6345495f3ffeHrPDfQKfwjff2i",
        "synthetic": true,
        "text": "<file>\n00001| \n</file>",
        "type": "text"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "part": {
        "filename": "a-empty.txt",
        "id": "prt_9cf8f64e0003v1TrnWYP1id0EW",
        "messageID": "msg_9cf8f64de0016tbfTQqWMydbdr",
        "mime": "text/plain",
        "sessionID": "ses_6345495f3ffeHrPDfQKfwjff2i",
        "type": "file",
        "url": "file:///Users/cam/tmp/a/a-empty.txt"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "info": {
        "cost": 0,
        "id": "msg_9cf8f6549001tpoRuqkwS4Rxtl",
        "mode": "build",
        "modelID": "claude-sonnet-4",
        "path": {
          "cwd": "/Users/cam/tmp/a",
          "root": "/Users/cam/tmp/a"
        },
        "providerID": "github-copilot",
        "role": "assistant",
        "sessionID": "ses_6345495f3ffeHrPDfQKfwjff2i",
        "system": {},
        "time": { "created": 1760123905353 },
        "tokens": {
          "cache": { "read": 0, "write": 0 },
          "input": 0,
          "output": 0,
          "reasoning": 0
        }
      }
    },
    "type": "message.updated"
  },
  {
    "properties": {
      "part": {
        "id": "prt_9cf8f701b001ojnIOsDWtIIrRM",
        "messageID": "msg_9cf8f6549001tpoRuqkwS4Rxtl",
        "sessionID": "ses_6345495f3ffeHrPDfQKfwjff2i",
        "type": "step-start"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "part": {
        "id": "prt_9cf8f7069001NijG6ReU9XlLwq",
        "messageID": "msg_9cf8f6549001tpoRuqkwS4Rxtl",
        "sessionID": "ses_6345495f3ffeHrPDfQKfwjff2i",
        "text": "1",
        "time": { "start": 1760123908201 },
        "type": "text"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "part": {
        "id": "prt_9cf8f7069001NijG6ReU9XlLwq",
        "messageID": "msg_9cf8f6549001tpoRuqkwS4Rxtl",
        "sessionID": "ses_6345495f3ffeHrPDfQKfwjff2i",
        "text": "1",
        "time": { "end": 1760123908202, "start": 1760123908202 },
        "type": "text"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "part": {
        "cost": 0,
        "id": "prt_9cf8f706a001BgsbkWs9dvxtt7",
        "messageID": "msg_9cf8f6549001tpoRuqkwS4Rxtl",
        "sessionID": "ses_6345495f3ffeHrPDfQKfwjff2i",
        "tokens": {
          "cache": { "read": 8435, "write": 0 },
          "input": 15511,
          "output": 5,
          "reasoning": 0
        },
        "type": "step-finish"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "info": {
        "cost": 0,
        "id": "msg_9cf8f6549001tpoRuqkwS4Rxtl",
        "mode": "build",
        "modelID": "claude-sonnet-4",
        "path": { "cwd": "/Users/cam/tmp/a", "root": "/Users/cam/tmp/a" },
        "providerID": "github-copilot",
        "role": "assistant",
        "sessionID": "ses_6345495f3ffeHrPDfQKfwjff2i",
        "system": {},
        "time": { "created": 1760123905353 },
        "tokens": {
          "cache": { "read": 8435, "write": 0 },
          "input": 15511,
          "output": 5,
          "reasoning": 0
        }
      }
    },
    "type": "message.updated"
  },
  {
    "properties": {
      "info": {
        "cost": 0,
        "id": "msg_9cf8f6549001tpoRuqkwS4Rxtl",
        "mode": "build",
        "modelID": "claude-sonnet-4",
        "path": { "cwd": "/Users/cam/tmp/a", "root": "/Users/cam/tmp/a" },
        "providerID": "github-copilot",
        "role": "assistant",
        "sessionID": "ses_6345495f3ffeHrPDfQKfwjff2i",
        "system": {},
        "time": { "completed": 1760123908223, "created": 1760123905353 },
        "tokens": {
          "cache": { "read": 8435, "write": 0 },
          "input": 15511,
          "output": 5,
          "reasoning": 0
        }
      }
    },
    "type": "message.updated"
  },
  {
    "properties": {
      "info": {
        "cost": 0,
        "id": "msg_9cf8f6549001tpoRuqkwS4Rxtl",
        "mode": "build",
        "modelID": "claude-sonnet-4",
        "path": { "cwd": "/Users/cam/tmp/a", "root": "/Users/cam/tmp/a" },
        "providerID": "github-copilot",
        "role": "assistant",
        "sessionID": "ses_6345495f3ffeHrPDfQKfwjff2i",
        "system": {},
        "time": { "completed": 1760123908224, "created": 1760123905353 },
        "tokens": {
          "cache": { "read": 8435, "write": 0 },
          "input": 15511,
          "output": 5,
          "reasoning": 0
        }
      }
    },
    "type": "message.updated"
  },
  {
    "properties": {
      "info": {
        "cost": 0,
        "id": "msg_9cf8f6549001tpoRuqkwS4Rxtl",
        "mode": "build",
        "modelID": "claude-sonnet-4",
        "path": { "cwd": "/Users/cam/tmp/a", "root": "/Users/cam/tmp/a" },
        "providerID": "github-copilot",
        "role": "assistant",
        "sessionID": "ses_6345495f3ffeHrPDfQKfwjff2i",
        "system": {},
        "time": { "completed": 1760123908224, "created": 1760123905353 },
        "tokens": {
          "cache": { "read": 8435, "write": 0 },
          "input": 15511,
          "output": 5,
          "reasoning": 0
        }
      }
    },
    "type": "message.updated"
  }
]
