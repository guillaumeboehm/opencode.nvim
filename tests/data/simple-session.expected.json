{"timestamp":1760658431,"lines":["","----","","","only answer the following, nothing else:","","1","","[a-empty.txt](a-empty.txt)","","----","","","1",""],"actions":[],"extmarks":[[1,2,0,{"virt_text_hide":false,"right_gravity":true,"virt_text_repeat_linebreak":false,"ns_id":3,"priority":10,"virt_text":[["▌󰭻 ","OpencodeMessageRoleUser"],[" "],["USER","OpencodeMessageRoleUser"],["","OpencodeHint"],[" (2025-10-10 19:18:25)","OpencodeHint"],[" [msg_9cf8f64de0016tbfTQqWMydbdr]","OpencodeHint"]],"virt_text_win_col":-3,"virt_text_pos":"win_col"}],[2,3,0,{"virt_text_hide":false,"right_gravity":true,"virt_text_repeat_linebreak":true,"ns_id":3,"priority":4096,"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_win_col":-3,"virt_text_pos":"win_col"}],[3,4,0,{"virt_text_hide":false,"right_gravity":true,"virt_text_repeat_linebreak":true,"ns_id":3,"priority":4096,"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_win_col":-3,"virt_text_pos":"win_col"}],[4,5,0,{"virt_text_hide":false,"right_gravity":true,"virt_text_repeat_linebreak":true,"ns_id":3,"priority":4096,"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_win_col":-3,"virt_text_pos":"win_col"}],[5,6,0,{"virt_text_hide":false,"right_gravity":true,"virt_text_repeat_linebreak":true,"ns_id":3,"priority":4096,"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_win_col":-3,"virt_text_pos":"win_col"}],[6,7,0,{"virt_text_hide":false,"right_gravity":true,"virt_text_repeat_linebreak":true,"ns_id":3,"priority":4096,"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_win_col":-3,"virt_text_pos":"win_col"}],[7,8,0,{"virt_text_hide":false,"right_gravity":true,"virt_text_repeat_linebreak":true,"ns_id":3,"priority":4096,"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_win_col":-3,"virt_text_pos":"win_col"}],[8,11,0,{"virt_text_hide":false,"right_gravity":true,"virt_text_repeat_linebreak":false,"ns_id":3,"priority":10,"virt_text":[[" ","OpencodeMessageRoleAssistant"],[" "],["BUILD","OpencodeMessageRoleAssistant"],[" claude-sonnet-4","OpencodeHint"],[" (2025-10-10 19:18:25)","OpencodeHint"],[" [msg_9cf8f6549001tpoRuqkwS4Rxtl]","OpencodeHint"]],"virt_text_win_col":-3,"virt_text_pos":"win_col"}]]}